lib/services/all_question_service.dart

```diff
@@ -1,50 +1,34 @@
 import 'package:project_eureka_flutter/models/question_model.dart';
+import 'dart:async';
+import 'dart:convert';
+import 'package:http/http.dart' as http;
 
 class AllQuestionService {
-  // Here Eureka will get data from the database.
-  // At this point it contains just dummy lists of questions for test purposes
-  Future<List> getQuestions() async {
-    List data = [];
-    // List of 2 dummy questions
-    List list1 = List.generate(2, (index) => "Message")
-        .map((val) => QuestionModel(
-              category: "Technology",
-              date: DateTime.now(),
-              status: 1,
-              visible: 1,
-              title: "Linux installation issue - usb not found",
-              description:
-                  "Hello. I have a Smartbuy 16Gb USB 2.0 flash drive (with new memory controller) that is not recognized in any Linux system, but on Windows it recognized and worked fine. When I connect it to the PC on Linux system, nothing happens",
-            ))
-        .toList();
 
-    // List of another 2 dummy questions (different)
-    List list2 = List.generate(2, (index) => "Message")
-        .map((val) => QuestionModel(
-              category: "Household",
-              date: DateTime.now(),
-              status: 0,
-              visible: 1,
-              title: "Windows Issue",
-              description: "Hello",
-            ))
-        .toList();
+  // GET
+  Future<List<QuestionModel>> getQuestions() async {
+    final response = await http.get(Uri.http('IP ADDRESS', '/v1/questions'));
 
-    // List of another 2 dummy questions (different)
-    List list3 = List.generate(2, (index) => "Message")
-        .map((val) => QuestionModel(
-              category: "Technology",
-              date: DateTime.now(),
-              status: 1,
-              visible: 1,
-              title: "iPhone not working",
-              description: "iphone stopped working",
-            ))
-        .toList();
+    if (response.statusCode == 200) {
+      // If the server did return a 200 OK response, then parse the JSON.
+      // Backend service returns a list of json instances, while we need a list of question objects.
+      // JSON parser will go through each json instance, transform it to a question object,
+      //   and append to the list that will be used in eureka_list_view
+      // Loop is reversed, as we want to sort the questions by date and show the most recent questions on top
+      //   (new questions are appending to the bottom of the database)
 
-    // Combined 6 dummy questions together
-    List list2list3 = new List.from(list2)..addAll(list3);
-    data = new List.from(list1)..addAll(list2list3);
-    return data;
+      final body = json.decode(response.body);
+      print("All questions were loaded");
+
+      List<QuestionModel> questions = new List();
+      for (var i = body.length - 1; i >= 0; i--) {
+        questions.add(QuestionModel.fromJson(body[i]));
+      }
```

Comment:
For some reason when adding a new item to the list and pulling down to update, the item is added out of order. 
![image](https://user-images.githubusercontent.com/36316432/111849069-fda22c80-88c9-11eb-815b-af11a2d00a30.png)

Reply:
As I mentioned in "what I did", I made it to be sorted by status. So when the new question is created through the app it will be given an Active status and will appear on top. If question is answered, status changes to closed and moves below the active questions. If you don't think it's a good idea, I can remove the sorting

Reply:
I see, to be honest, I think the homepage should not display inactive questions. But I not sure if this the right approach. If the homepage has inactive questions, it can really clutter it up I think.

Reply:
In this case Search Bar won't be as useful, as the user may search for answers among inactive questions. There could be a checkbox in a filter menu that would display or hide inactive questions. And when search bar filter is applied, it will turn on the checkbox to display inactive questions by default for convenience. This may be discussed later, but for now I would leave it with just status ordering.